### PR TITLE
Second chance for user. First approach

### PR DESCRIPTION
**Pull request by JTrillo**
Pull request for issue [3](https://github.com/Lemoncode/english-quiz/issues/3), provide a second chance for users.
If you have any doubts or you think I should change anything, please tell me.

### CODE REVIEW

File: src/pods/test-verb-forms/test-verb-forms.business.ts [hunk 1]
```diff
@@ -33,13 +33,26 @@ const quizToLower = (quiz: VerbQuiz) => ({
...
+  let hint = "";
+  if (!verbCorrect.infinitive) hint+="infinitive, ";
+  if (!verbCorrect.past) hint+="past, ";
+  if (!verbCorrect.participle) hint+="participle";
+
+  // Trim the last part if it ends on ", "
+  if (hint[hint.length-2] === ',') hint = hint.substring(0, hint.length-2);
+
+  return hint;
 };
```
Comment (brauliodiez) [inline src/pods/test-verb-forms/test-verb-forms.business.ts:58]: Rather generate an enum, so we can handle it as well in the textField

File: src/pods/test-verb-forms/test-verb-forms.component.tsx [hunk 2]
```diff
@@ -66,9 +80,9 @@ export const TestVerbFormComponent: React.FC<Props> = props => {
                 <TextFieldComponent name="participle" label="Participle" />
               </div>
             )}
-            {validated ? (
+            {validated && (secondAttempt || verbCorrect.all) ? (
               <>
-                <ShowResults succeeded={isCorrect} verb={verb} />
+                <ShowResults secondAttempt={true} verbCorrect={verbCorrect} verb={verb} />
```
Comment (brauliodiez) [inline src/pods/test-verb-forms/test-verb-forms.component.tsx:85]: I guess this is true waiting for the setting to be added.

Right now add a TODO indicating that

```jsx
{/*TODO: pending on secondAttempt config setting*/}
```

File: src/pods/test-verb-forms/test-verb-forms.business.ts (hunk 1 above)
Comment (JTrillo) [reply src/pods/test-verb-forms/test-verb-forms.business.ts:58]: Ok. I will change the method.

File: src/pods/test-verb-forms/test-verb-forms.component.tsx (hunk 2 above)
Comment (JTrillo) [reply src/pods/test-verb-forms/test-verb-forms.component.tsx:85]: Yes, it is always true until we have implemented the settings storage.

File: src/pods/test-verb-forms/components/show-results.component.tsx [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
 export const ShowResults: React.FC<Props> = props => {
-  const { succeeded, verb } = props;
-  return succeeded ? (
+  const { secondAttempt, verbCorrect, verb } = props;
+  return verbCorrect.all ? (
     <span>RIGHT !!!!</span>
+  ) : !secondAttempt ? (
+    <span>
+      You have a second chance. You failed in {generateHint(verbCorrect)}.
+    </span>
```
[the remark this answers is not in the document]
Comment (JTrillo) [reply src/pods/test-verb-forms/components/show-results.component.tsx:18]: I see. I will make this change and the others you propose during the weekend.

**Comment by JTrillo** [comment]
I made all changes you suggested. Now, when second chance is enabled, incorrect fields will be highlighted (the message saying which fields are wrong is still displayed).

I also noticed that test starts with question 0, so the student will have an additional question than configured. I changed `INITIAL_CURRENT_QUESTION` from 0 to 1 in file `test-verb-forms.container.tsx`.

I detected another undesired behaviour: the last two questions were always the same. This happened because the second `useEffect` in file `test-verb-forms.container.tsx` did not change the verb for the last question.

If you have any doubts or you think I should change anything, please tell me.

**Pull request merged**